common/src/main/java/com/wynntils/features/user/overlays/map/MinimapFeature.java

```diff
@@ -205,47 +247,81 @@ public void render(PoseStack poseStack, float partialTicks, Window window) {
                 }
             }
 
-            // TODO compass icon
+            // Compass icon
+            Optional<WaypointPoi> compassOpt = CompassModel.getCompassWaypoint();
 
-            // cursor
-            if (!followPlayerRotation) {
-                poseStack.pushPose();
-                RenderUtils.rotatePose(
-                        poseStack, centerX, centerZ, 180 + McUtils.player().getYRot());
-            }
+            if (compassOpt.isEmpty()) return;
 
-            MapRenderer.renderCursor(
-                    poseStack, centerX, centerZ, this.pointerScale, this.pointerColor, this.pointerType);
+            WaypointPoi compass = compassOpt.get();
 
-            if (!followPlayerRotation) {
-                poseStack.popPose();
+            Pair<Float, Float> compassLocation = getRenderLocation(
+                    compass, playerX, playerZ, centerX, centerZ, sinRotationRadians, cosRotationRadians);
+            float compassRenderX = compassLocation.a();
+            float compassRenderZ = compassLocation.b();
+
+            final float compassSize = Math.max(compass.getWidth(), compass.getHeight()) * 0.8f * poiScale;
+
+            float compassOffsetX = compassRenderX - centerX;
+            float compassOffsetZ = compassRenderZ - centerZ;
+            float distance = MathUtils.magnitude(compassOffsetX, compassOffsetZ);
+
+            compassOffsetX /= distance;
+            compassOffsetZ /= distance;
+
+            float scaledWidth = width - 4 * compassSize;
+            float scaledHeight = height - 4 * compassSize;
+
+            float toSquareScale = 1f;
+
+            if (maskType == MapMaskType.Rectangular) {
+                // Scale as necessary
+                toSquareScale =
+                        Math.min(scaledWidth / Math.abs(compassOffsetX), scaledHeight / Math.abs(compassOffsetZ)) / 2;
+            } else if (maskType == MapMaskType.Circle) {
+                toSquareScale = scaledWidth
+                        / (MathUtils.magnitude(compassOffsetX, compassOffsetZ * scaledWidth / scaledHeight))
```

Comment:
feels a bit weird to call it like this here

Reply:
Call what?

Reply:
toSquareScale when map is circle?

Reply:
Oops